### PR TITLE
fix(clapi) RT acknowledge/downtime

### PR DESCRIPTION
**Pull request by ghost**
## Description

Jira : MON-14791

> Creation of downtimes/acknowlegments may fail silently when object in parameters is invalid
> Action should return 'object not found' error when object is not found

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x
- [x] 22.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by callapa — DISMISSED** [review]

**Review by kduret — DISMISSED** [review]

File: www/class/centreon-clapi/Validator/RtValidator.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    private $serviceObject;
+
+    public function __construct(CentreonHost $hostObject, CentreonService $serviceObject)
+    {
+        $this->hostObject = $hostObject;
+        $this->serviceObject = $serviceObject;
+    }
+
+    /**
+     * Mysql perform an insensitive case search. We want to enforce the strictness here.
```
Comment (jeremyjaouen) [inline www/class/centreon-clapi/Validator/RtValidator.php]:
```suggestion
     * Data storage perform an insensitive case search. We want to enforce the strictness here.
```
we should avoid to mentions the type of datastorage

File: www/class/centreon-clapi/Validator/RtValidator.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        $realHostName = $this->hostObject->getHostName($hostIdAndServiceId[0]);
+        $realServiceName = $this->serviceObject->getObjectName($hostIdAndServiceId[1]);
+
+        return $realHostName === $host
+            && $realServiceName === $service
+            && $this->serviceObject->serviceExists($host, $service);
+    }
+
+    /**
+     * Mysql perform an insensitive case search. We want to enforce the strictness here.
```
Comment (jeremyjaouen) [inline www/class/centreon-clapi/Validator/RtValidator.php]:
```suggestion
     * Data storage perform an insensitive case search. We want to enforce the strictness here.
```

**Review by jeremyjaouen — APPROVED** [review]

**Review by wtermellil — APPROVED** [review]

**Comment by kduret** [comment]
migrated to https://github.com/centreon/centreon-gha/pull/143

**Comment by ghost** [comment]
👉️ This PR is close because of the **centreon-gha** migration

**Pull request closed without merging**